test/e2e/framework/resource_usage_gatherer.go

```diff
@@ -263,18 +265,37 @@ func NewResourceUsageGatherer(c *client.Client, options ResourceGathererOptions)
 			return nil, err
 		}
 
-		g.workerWg.Add(len(nodeList.Items))
-		for _, node := range nodeList.Items {
-			g.workers = append(g.workers, resourceGatherWorker{
-				c:                    c,
-				nodeName:             node.Name,
-				wg:                   &g.workerWg,
-				containerIDToNameMap: g.containerIDToNameMap,
-				containerIDs:         g.containerIDs,
-				stopCh:               g.stopCh,
-				finished:             false,
-				inKubemark:           false,
-			})
+		if options.masterOnly {
```

Comment:
I'm not sure it'll increase the readability. Note that also a waitGroup changes. I can wrap the append call in the function if you think it'll help with something.